### PR TITLE
[WIP] File size

### PR DESCRIPTION
**Pull request by jbaxleyiii**
This PR seeks to lower the filesize of the generated library as much as possible and add limits to the build process

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -9,7 +9,6 @@
     "rootDir": ".",
     "outDir": "lib",
     "allowSyntheticDefaultImports": true,
-    "pretty": true,
```
Comment (stubailo) [inline tsconfig.json]: This shouldn't really affect the output size right? I bet uglify and gzip do basically the same thing?

Comment (jbaxleyiii) [inline tsconfig.json]: Nope! Pretty much every build tool an app would use would minify so even if we don't, it shouldn't affect it

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+var gulp = require('gulp'),
```
Comment (stubailo) [inline gulpfile.js]: I feel like this script should go in some sort of `scripts/` directory --- having it in the root implies that we use it to build stuff.

Comment (jbaxleyiii) [inline gulpfile.js]: Tbh, Id rather rewrite all of this! I've got some ideas and want it to fail a build if it gets above a threshold. Essentially what I have now was a test / proof of concept

Comment (stubailo) [inline gulpfile.js]: Oh, OK! Ignore me then :]

**Comment by jbaxleyiii** [comment]
@stubailo I'm going to close this for now. When we get to the alpha, I'll take that code base and see how small we can get it :+1:

**Comment by stubailo** [comment]
@jbaxleyiii we might want to do some of the super obvious ones like the single-package lodash stuff though.

**Comment by jbaxleyiii** [comment]
@stubailo absolutely! I have to write the type defs for those. I'm still working on this thinning everything just didn't want the long open PR with master changing so much :+1:

**Pull request closed without merging**